binderhub/tests/test_app.py

```diff
@@ -2,10 +2,33 @@
 
 from subprocess import check_output
 import sys
+import pytest
+
+from traitlets import TraitError
+
+from binderhub.app import BinderHub
+from binderhub.repoproviders import (RepoProvider, GitLabRepoProvider, GitHubRepoProvider)
+
 
 def test_help():
     check_output([sys.executable, '-m', 'binderhub', '-h'])
 
 def test_help_all():
     check_output([sys.executable, '-m', 'binderhub', '--help-all'])
 
+def test_repo_providers():
+    b = BinderHub()
+
+    class Provider(RepoProvider):
+        pass
+
+    b.repo_providers = dict(gh=GitHubRepoProvider, gl=GitLabRepoProvider)
+    b.repo_providers = dict(p=Provider)
```

Comment:
This doesn't look right, `b.repo_providers` is set twice.

Reply:
And as soon as I posted that comment I realised what it's doing- it's checking the `repo_providers` property is validated. Maybe add a comment to help others?

Reply:
Thanks @manics!